cmake/profiles/common.cmake

```diff
@@ -0,0 +1,136 @@
+if(MBED_TOOLCHAIN strequal "GCC_ARM")
+
+set(COMMON_FLAGS
+    -Wall
+    -Wextra
+    -Wno-unused-parameter
+    -Wno-missing-field-initializers
+    -fmessage-length=0
+    -fno-exceptions
+    -ffunction-sections
+    -fdata-sections
+    -funsigned-char
+    -MMD
+    -fomit-frame-pointer
+    -Os
+    -g
+)
+set(CMAKE_C_FLAGS ${CMAKE_C_FLAGS}
+    ${COMMON_FLAGS}
+)
+set(CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS}
+    ${COMMON_FLAGS}
+)
+set(CMAKE_ASM_FLAGS ${CMAKE_ASM_FLAGS}
+    ${COMMON_FLAGS}
+)
+set(CMAKE_CXX_LINK_FLAGS ${CMAKE_CXX_LINK_FLAGS}
+    ${COMMON_FLAGS}
+)
+
+elseif(MBED_TOOLCHAIN strequal "ARMC6")
+
+set(COMMON_FLAGS
+    -c
+    --target=arm-arm-none-eabi
+    -mthumb
```

Comment:
We may have specified mthumb before here, but probably this should just come from the core only. This isn't an option we'd want to use on A-class targets, for example. Can we remove this option in a separate commit?